tests/functional/conftest.py

```diff
@@ -17,15 +17,6 @@ def auto_use_dn(db_engine):
     return db_engine
 
 
-def _clean_database(engine):
-    tables = reversed(db.BASE.metadata.sorted_tables)
-    with contextlib.closing(engine.connect()) as conn:
-        tx = conn.begin()
-        tnames = ", ".join('"' + t.name + '"' for t in tables)
-        conn.execute("TRUNCATE {};".format(tnames))
-        tx.commit()
```

Comment:
:wave:

Reply:
I've put this back but as an `autouse` fixture rather than a helper function that the `app` fixture calls because we want the DB to be cleaned before every test, not just before tests that use the `app` fixture. And because cleaning the DB doesn't have anything to do with creating the WSGI app and shouldn't be mixed in with it. Also moved the `clean_database` fixture to the top of `conftest.py` so that it executes before any other fixtures.